libssu/ssudeviceinfo.cpp

```diff
@@ -413,39 +413,42 @@ QStringList SsuDeviceInfo::repos(bool rnd, int filter)
     int adaptationCount = adaptationRepos().size();
     QStringList result;
 
-
-    ///@TODO move this to a hash, containing repo and enabled|disabled
-    ///      write repos with enabled/disabled to disks
-    ///      for the compat functions providing a stringlist, do the filtering
-    ///      run only when creating the list, based on the enabled|disabled flags
     if (filter == Ssu::NoFilter ||
             filter == Ssu::BoardFilter ||
-            filter == Ssu::BoardFilterUserBlacklist) {
+            filter == Ssu::BoardFilterUserBlacklist ||
+            filter == Ssu::BoardFilterAvailable) {
+        SsuSettings repoSettings(SSU_REPO_CONFIGURATION, SSU_REPO_CONFIGURATION_DIR);
         // for repo names we have adaptation0, adaptation1, ..., adaptationN
         for (int i = 0; i < adaptationCount; i++)
             result.append(QString("adaptation%1").arg(i));
 
-        // now read the release/rnd repos
-        SsuSettings repoSettings(SSU_REPO_CONFIGURATION, SSU_REPO_CONFIGURATION_DIR);
-        QString repoKey = (rnd ? "default-repos/rnd" : "default-repos/release");
-        if (repoSettings.contains(repoKey))
-            result.append(repoSettings.value(repoKey).toStringList());
-
-        // TODO: add specific repos (developer, sdk, ..)
-
-        // add device configured repos
-        if (boardMappings->contains(deviceVariant(true) + "/repos"))
-            result.append(boardMappings->value(deviceVariant(true) + "/repos").toStringList());
-
-        // add device configured repos only valid for rnd and/or release
-        repoKey = (rnd ? "/repos-rnd" : "/repos-release");
-        if (boardMappings->contains(deviceVariant(true) + repoKey))
-            result.append(boardMappings->value(deviceVariant(true) + repoKey).toStringList());
-
-        // read the disabled repositories for this device
-        // user can override repositories disabled here in the user configuration
-        foreach (const QString &key, disabledRepos())
-            result.removeAll(key);
+        if (filter == Ssu::BoardFilterAvailable) {
+            QString group = (rnd ? "rnd" : "release");
+            repoSettings.beginGroup(group);
+            result.append(repoSettings.allKeys());
+            repoSettings.endGroup();
+        } else {
+            // now read the release/rnd repos
+            QString repoKey = (rnd ? "default-repos/rnd" : "default-repos/release");
+            if (repoSettings.contains(repoKey))
+                result.append(repoSettings.value(repoKey).toStringList());
+
+            // TODO: add specific repos (developer, sdk, ..)
+
+            // add device configured repos
```

Comment:
Should these two sections be moved to before the if (filter == Ssu::BoardFilterAvailable)? I don't think we set any repos this way right now, but if we did they they should count as available too?